src/components/pricing/pricing-widget/index.tsx

```diff
@@ -71,6 +73,26 @@ const PricingWidget: FunctionComponent<{}> = () => {
     })
 
     if (emailIsValid(viewer?.email)) {
+      const {hasProAccess} = await axios
+        .post(`/api/users/check-pro-status`, {
+          email: viewer?.email,
+        })
+        .then(({data}) => data)
+
+      if (hasProAccess) {
+        const message = `You already have pro access with this account (${viewer?.email}). Please contact support@egghead.io if you need help with your subscription.`
+
+        toast.error(message, {
+          duration: 6000,
+          icon: '❌',
+        })
+
+        // email is already associated with a pro account, return early instead
+        // of sending the user to a Stripe Checkout Session.
+        return
+      }
+
+      // the user doesn't have pro access, proceed to checkout
```

Comment:
the line below was already here but track calls shouldn’t be blocking in general but i’m not looking at all the context so might be required here